dex/testing/eth/harness.sh

```diff
@@ -0,0 +1,139 @@
+# "extradata". The addresses in the "alloc" field are allocated "balance".
+cat > "${NODES_ROOT}/genesis.json" <<EOF
+{
+  "config": {
+    "chainId": 42,
+    "homesteadBlock": 0,
+    "eip150Block": 0,
+    "eip155Block": 0,
+    "eip158Block": 0,
+    "byzantiumBlock": 0,
+    "constantinopleBlock": 0,
+    "petersburgBlock": 0,
+    "clique": {
+      "period": 1,
+      "epoch": 30000
+    }
+  },
+  "difficulty": "1",
+  "gasLimit": "8000000",
+  "extradata": "0x00000000000000000000000000000000000000000000000000000000000000009ebba10a6136607688ca4f27fab70e23938cd0270000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+  "alloc": {
+    "18d65fb8d60c1199bb1ad381be47aa692b482605": {
+        "balance": "1000000000000"
+    },
+    "4f8ef3892b65ed7fc356ff473a2ef2ae5ec27a06": {
+        "balance": "1000000000000"
+    }
+  }
+}
+EOF
+
+# Write mining javascript.
+# NOTE: This sometimes mines more than one block. It is a race.
+cat > "${MINE_JS}" <<EOF
+function mine() {
+  miner.start();
+  miner.stop();
+  admin.sleep(1.1);
+  return true;
+}
+EOF
+
+# Add wallet script.
+HARNESS_DIR=$(dirname "$0")
+cp "${HARNESS_DIR}/create-node.sh" "${NODES_ROOT}/harness-ctl/create-node"
+
+# Reorg script
+# TODO: Make this.
+
+# Shutdown script
+cat > "${NODES_ROOT}/harness-ctl/quit" <<EOF
+#!/bin/sh
+tmux send-keys -t $SESSION:1 C-c
+tmux send-keys -t $SESSION:2 C-c
+tmux kill-session
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/quit"
+
+################################################################################
+# Start harness
+################################################################################
+
+echo "Starting harness"
+tmux new-session -d -s $SESSION
+tmux rename-window -t $SESSION:0 'harness-ctl'
+tmux send-keys -t $SESSION:0 "set +o history" C-m
+tmux send-keys -t $SESSION:0 "cd ${NODES_ROOT}/harness-ctl" C-m
+
+################################################################################
+# Eth nodes
+################################################################################
+
+echo "Starting simnet alpha node"
+"${HARNESS_DIR}/create-node.sh" "$SESSION:1" "alpha" "$ALPHA_NODE_PORT " \
+	"$CHAIN_ADDRESS" "$PASSWORD" "$CHAIN_ADDRESS_JSON " \
+	"$CHAIN_ADDRESS_JSON_FILE_NAME" "$ALPHA_ADDRESS" "$PASSWORD" \
+	"$ALPHA_ADDRESS_JSON " "$ALPHA_ADDRESS_JSON_FILE_NAME" "$ALPHA_NODE_KEY"
+
+echo "Starting simnet beta node"
+"${HARNESS_DIR}/create-node.sh" "$SESSION:2" "beta" "$BETA_NODE_PORT " \
+	"$CHAIN_ADDRESS" "$PASSWORD" "$CHAIN_ADDRESS_JSON " \
+	"$CHAIN_ADDRESS_JSON_FILE_NAME" "$BETA_ADDRESS" "$PASSWORD" \
+	"$BETA_ADDRESS_JSON " "$BETA_ADDRESS_JSON_FILE_NAME" "$BETA_NODE_KEY"
+
+# NOTE: This will cause beta to connect automatically to alpha.
+echo "Connecting nodes"
+"${NODES_ROOT}/harness-ctl/alpha" "attach --exec admin.addPeer('enode://${BETA_ENODE}@127.0.0.1:$BETA_NODE_PORT')"
```

Comment:
Maybe this needs a sleep in-between starting and connecting, but this is working fine for me.